app/models/compute_resources/foreman/model/openstack.rb

```diff
@@ -127,6 +127,14 @@ def user_data_supported?
       true
     end
 
+    def zones
+      @zones ||= begin
+        client.list_zones.data[:body]["availabilityZoneInfo"].map{|i| i["zoneName"]}
+      rescue
```

Comment:
this is, rescue also saves from client.list_zones.data[:body]["availabilityZoneInfo"] being nil (in which case map will throw an error)
so rescue fog + no method error?